diff -urN a/config.sub b/config.sub
--- a/config.sub	2019-10-27 13:14:58.866612800 -0300
+++ b/config.sub	2019-10-27 14:05:33.235840000 -0300
@@ -1341,7 +1341,7 @@
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
-	     | aos* | aros* | cloudabi* | sortix* \
+	     | aos* | aros* | cloudabi* | sortix* | chicago* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | knetbsd* | mirbsd* | netbsd* \
--- a/fixincludes/mkfixinc.sh	2019-10-27 13:14:39.335172400 -0300
+++ b/fixincludes/mkfixinc.sh	2019-10-27 14:17:09.667824600 -0300
@@ -11,8 +11,10 @@
 
 # Check for special fix rules for particular targets
 case $machine in
+    i?86-*-chicago* | \
     i?86-*-cygwin* | \
     i?86-*-mingw32* | \
+    x86_64-*-chicago* | \
     x86_64-*-mingw32* | \
     powerpc-*-eabisim* | \
     powerpc-*-eabi*    | \
--- a/gcc/config/chicago.h	1969-12-31 21:00:00.000000000 -0300
+++ b/gcc/config/chicago.h	2019-10-27 14:18:25.141363500 -0300
@@ -0,0 +1,31 @@
+#undef TARGET_CHICAGO
+#define TARGET_CHICAGO 1
+
+#undef STANDARD_STARTFILE_PREFIX
+#define STANDARD_STARTFILE_PREFIX "/System/Libraries/ /Development/Libraries/"
+
+#undef LIB_SPEC
+#define LIB_SPEC "-lc" /* link everything against the C standard library */
+
+#undef LIBSTDCXX
+#define LIBSTDCXX "cxx" /* CHicago's c++ library is called libcxx (the gcc's default is called libstdc++) */
+
+/* Files that are linked before user code.
+   The %s tells GCC to look for these files in the library directory. */
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared:crt0.o%s} crti.o%s %{!shared:crtbegin.o%s}"
+
+/* Files that are linked after user code. */
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "%{!shared:crtend.o%s} crtn.o%s"
+
+#undef LINK_SPEC
+#define LINK_SPEC "%{shared:-shared} %{static:-static} %{!shared: %{!static: %{rdynamic:-export-dynamic}}} -z max-page-size=4096"
+
+/* Don't automatically add extern "C" { } around header files. */
+#undef  SYSTEM_IMPLICIT_EXTERN_C
+#define SYSTEM_IMPLICIT_EXTERN_C 0
+
+/* Additional predefined macros. */
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() do { builtin_define ("__chicago__"); builtin_assert ("system=chicago"); } while(0);
--- a/gcc/config.gcc	2019-10-27 13:14:44.579559400 -0300
+++ b/gcc/config.gcc	2019-10-27 14:09:37.451824900 -0300
@@ -675,6 +675,13 @@
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-chicago*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  use_gcc_stdint=wrap
+  native_system_header_dir=/Development/Headers
+  ;;
 *-*-darwin*)
   tmake_file="t-darwin "
   tm_file="${tm_file} darwin.h"
@@ -1609,6 +1616,12 @@
 		dwarf2=no
 	fi
 	;;
+i[34567]86-*-chicago*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h chicago.h"
+	;;
+x86_64-*-chicago*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h i386/x86-64.h chicago.h"
+	;;
 i[34567]86-*-darwin1[89]*)
 	echo "Error: 32bit target is not supported after Darwin17" 1>&2
 	;;
--- a/libgcc/config.host	2019-10-27 13:14:39.665175000 -0300
+++ b/libgcc/config.host	2019-10-27 14:15:50.173730100 -0300
@@ -616,6 +616,14 @@
 hppa*-*-openbsd*)
 	tmake_file="$tmake_file pa/t-openbsd"
 	;;
+i[34567]86-*-chicago*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
+x86_64-*-chicago*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 i[34567]86-*-darwin*)
 	tmake_file="$tmake_file i386/t-crtpc t-crtfm i386/t-msabi"
 	tm_file="$tm_file i386/darwin-lib.h"
